src/main/java/io/jenkins/plugins/monitoring/Monitor.java

```diff
@@ -0,0 +1,179 @@
+package io.jenkins.plugins.monitoring;
+
+import com.google.common.collect.ImmutableSet;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import io.jenkins.plugins.monitoring.util.PortletService;
+import io.jenkins.plugins.monitoring.util.PullRequestFinder;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import org.jenkinsci.plugins.workflow.steps.SynchronousStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.io.Serializable;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * This {@link Step} is responsible for the configuration of the monitoring dashboard
+ * via the corresponding Jenkinsfile.
+ *
+ * @author Simon Symhoven
+ */
+public final class Monitor extends Step implements Serializable {
+    private static final long serialVersionUID = -1329798203887148860L;
+    private String portlets;
+
+    /**
+     * Creates a new instance of {@link Monitor}.
+     *
+     * @param portlets
+     *              the monitor configuration as json array string.
+     */
+    @DataBoundConstructor
+    public Monitor(final String portlets) {
+        super();
+        this.portlets = portlets;
+    }
+
+    public void setPortlets(String portlets) {
+        this.portlets = portlets;
+    }
```

Comment:
Yes it is required. The ports json string is set in the constructor, but if the user provides an invalid portlet-id inside this configuration I have to remove the invalid one and set the validated json string again to prevent the run to fail. But I change the visibility of the setter to private now.

Reply:
or how would you solve it?